tests/regression/27-inv_invariants/22-meet-ptrs.c

```diff
@@ -0,0 +1,24 @@
+//PARAM:  --enable ana.int.interval
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+
+int main() {
+   int arr[20];
+
+   int top;
+
+   int i = 2;
+   if(top) {
+      i = 8;
+   }
+
+   int* imprecise = &arr[i];
+
+   if(imprecise == &arr[2]) {
+      __goblint_check(imprecise == &arr[2]);
```

Comment:
Maybe this didn't work before, but if I understand the improvement correctly, this is a rather pedestrian case of it.
We should now be able to refine `{&a[0..1]} == {&a[1..2]}` into `{&a[1]}` too, right?

Reply:
I added a more intricate example, though refinement currently happens only in one direction, which we should probably investigate separately.